### PR TITLE
[53,133,141,143,145,147,148] A little more TLC for PREP

### PR DESCRIPTION
**Pull request by SwooshyCueb**
Addresses #133
Addresses #141
Addresses #143
Addresses #145
Addresses #147
Addresses #148

I was also going to include 142 in this PR, but I need to get back to working on the development environment and audit, so I'm shelving that one for now.

### CODE REVIEW

**Review by korydraughn — COMMENTED** [review]
Everything looks good so far.

I wonder if we should add a document for developer notes? The purpose being to hold info that doesn't have a good place to live in the source. This doc could hold anything.

Thoughts?

**Comment by trel** [comment]
> I wonder if we should add a document for developer notes?

Almost a philosophy -> execution / planning document.

I would like that.    hmm, design.md ?

**Comment by korydraughn** [comment]
I see some projects have a file named `DEVELOP.md` or something along those lines. I'm not a huge fan of that name, but it gets the point across fairly easily.

Willing to talk more about conventions for all projects.

File: CMakeLists.txt [hunk 1]
```diff
@@ -41,6 +41,37 @@ find_package(OpenSSL REQUIRED COMPONENTS Crypto SSL)
...
+      if (Python_VERSION_MAJOR EQUAL "3" AND Python_VERSION_MINOR LESS "8")
+        set(__PYTHON_VERSION_SUFFIX "m")
+      else()
+        set(__PYTHON_VERSION_SUFFIX "")
+      endif()
+      set(IRODS_PYTHON_SONAME "${CMAKE_SHARED_LIBRARY_PREFIX}python${Python_VERSION_MAJOR}.${Python_VERSION_MINOR}${__PYTHON_VERSION_SUFFIX}${CMAKE_SHARED_LIBRARY_SUFFIX}.1.0")
+    endif()
+  endif()
+  message("Python SONAME detected as ${IRODS_PYTHON_SONAME}")
+endif()
```
[the remark this answers is not in the document]
Comment (korydraughn) [reply CMakeLists.txt]: Gotcha.

**Review by alanking — COMMENTED** [review]
Everything looks fine to me. Agreed on adding those "why" comments

**Comment by korydraughn** [comment]
Looking ...

**Review by korydraughn — COMMENTED** [review]
Everything looks good.

I take it that nothing has changed since the last review that requires rerunning the test suite?

**Comment by SwooshyCueb** [comment]
> I take it that nothing has changed since the last review that requires rerunning the test suite?

Nah, these changes are pretty trivial. I don't think we need to re-run the test suite. We'll be running it again anyway for my next PR.

**Review by korydraughn — APPROVED** [review]
@SwooshyCueb If you're confident/happy with the changes, add the pounds and let's get it in.

**Comment by SwooshyCueb** [comment]
\#'d

**Review by alanking — APPROVED** [review]

**Comment by korydraughn** [comment]
> Everything looks good so far.
> 
> I wonder if we should add a document for developer notes? The purpose being to hold info that doesn't have a good place to live in the source. This doc could hold anything.
> 
> Thoughts?

I'm going to capture this for later discush. This idea likely applies to other repos too.

**Pull request merged**